pkg/prometheus/statefulset.go

```diff
@@ -96,27 +96,41 @@ func prometheusNameByShard(name string, shard int32) string {
 func makeStatefulSet(
 	logger log.Logger,
 	name string,
-	p monitoringv1.Prometheus,
+	baseImage, tag, sha string,
```

Comment:
(nit) I'd pass a fully constructed image location to the function. It would be up to the caller (e.g. pkg/prometheus/operator.go) to build it as it needs.

Reply:
Nice, that sounds like a good approach!

Reply:
Actually, this is a bit harder than I thought when I started implementing it. `p.Spec.Image` is a string pointer instead of a single string.

A ton of tests started to fail with nil pointer exceptions and I could find a good approach to make them pass 😅.

Another strategy I tried was changing the signature of `operator.BuildImagePath` to return `*string` instead of `string` but, that requires adding `pointer.String(...)` to a lot of returns... not sure what is more readable 😬